config/config.go

```diff
@@ -379,6 +379,21 @@ func (c *Config) unmarshalRemotes() error {
 		c.Remotes[r.Name] = r
 	}
 
+	// Check if the main remote section still has options
+	// after unmarshaling named subsections - this indicates an empty subsection name
+	if len(s.Options) > 0 {
+		emptySubsection := &format.Subsection{
+			Name:    "",
+			Options: s.Options,
+		}
+
+		r := &RemoteConfig{}
+		if err := r.unmarshal(emptySubsection); err != nil {
+			return err
+		}
+
+		c.Remotes[r.Name] = r
+	}
```

Comment:
Would this not lead to options of a different subsections to end up in the empty remote section? Please add tests to that effect. We probably need at least two test cases, one that the first remote is unnamed and the second isn't, and the next which is revered. They should have different options so that we can verify we are loading the correct values.

Assuming this works as intended, we probably could remove this `if` block and amend the original `for` to:

```golang
	for _, sub := range s.Subsections {
		r := &RemoteConfig{}
		if len(s.Options) > 0 {
			sub.Options = s.Options
		}

		if err := r.unmarshal(sub); err != nil {
			return err
		}
		c.Remotes[r.Name] = r
	}
```